htdocs/projet/class/api_projects.class.php

```diff
@@ -82,6 +84,64 @@ public function get($id)
 		return $this->_cleanObjectDatas($this->project);
 	}
 
+	/**
+	 * Get contacts of a project.
+	 *
+	 * @param int   $id                     Id of project
+	 * @return  array                               Array of contacts objects
+	 *
+	 * @url	GET {id}/contacts
+	 * @throws 	RestException
+	 */
+	public function getContacts($id)
+	{
+		global $db, $conf;
+
+                // No need to check rights for the users when using User API because it is already in the get method
+		if ( (!DolibarrApiAccess::$user->rights->projet->lire) && (!DolibarrApiAccess::$user->rights->contact->lire) ){
+			throw new RestException(401);
+		}
+
+		$obj_ret = array();
+
+		$sql = "SELECT fk_socpeople, fk_c_type_contact";
+		$sql .= " FROM ".MAIN_DB_PREFIX."element_contact";
+		$sql .= " WHERE element_id=".$id;
+
+                // Add internal and external people for project type of contact
+                $sql .= ' AND (fk_c_type_contact=160 OR fk_c_type_contact=161 OR fk_c_type_contact=170 OR fk_c_type_contact=171)';
+
+		dol_syslog("API Rest request");
+		$result = $this->db->query($sql);
+
+		if ($result) {
+                        foreach ($result as $key => $value) {
+                            // fk_c_type_contact = 160, 161 we use users object because they are internal contacts
+                            // fk_c_type_contact = 170, 171 we use contacts object because they are external contacts
+                            if ($value["fk_c_type_contact"] == "160" || $value["fk_c_type_contact"] == "161") {
```

Comment:
Hard coded value 160 and 161 are not allowed.
Take a look at other API getContacts for other objects like api_orders.php to make a similar code...

Reply:
Hello @eldy 

I don't know the complete code of Dolibarr. If you mean something like that

```
		// Add external contacts ids
		$tmparray = $this->commande->liste_contact(-1, 'external', $contact_list);
		if (is_array($tmparray)) {
			$this->commande->contacts_ids = $tmparray;
		}
		$this->commande->fetchObjectLinked();
```

It would be faster to tell me what to change and how.

Regards,

Reply:
Yes, i speak about this example. Just replace  $this->commande with this->project

Reply:
Hello @eldy 

I thought I would have time to make the needed change, but it's not the case...
It would also take more time because when I will have time, I would need to dig again in the code to make the change properly.

If you have spare to do it, I guess it will be faster.
If not, I keep it on my to-do list when I have time for it.

Cheers

Reply:
Hello @eldy 

Still no time to dig again where to add the example above, but in our use case, it is interesting to be able to see the `$object->array_options` so I think it would be ok to comment `unset($object->array_options);`.

I wanted to notify you in case you had time to make the update.

Cheers